ivy/functional/ivy/gradients.py

```diff
@@ -284,21 +299,26 @@ def optimizer_update(ws, effective_grads, lr, inplace=None, stop_gradients=True)
 
     """
     inplace = ivy.default(inplace, ivy.inplace_variables_supported())
-    layerwise_lr = isinstance(lr, ivy.Container)
-    deltas = effective_grads.map(
-        lambda eff_grad, kc: ((lr[kc] if layerwise_lr else lr) * eff_grad)
-    )
+    deltas = effective_grads * lr
     if inplace:
-        ws = ws.map(lambda w, kc: ivy.inplace_decrement(w, deltas[kc]))
+        ws = ivy.inplace_decrement(ws, deltas)
     else:
-        ws = ws.map(lambda w, kc: -deltas[kc] + w)
+        ws = ws - deltas
+
     if stop_gradients:
-        return ws.stop_gradients(preserve_type=True)
+        return stop_gradient(ws, preserve_type=True)
     return ws
 
 
 @to_native_arrays_and_back
-def gradient_descent_update(ws, dcdws, lr, inplace=None, stop_gradients=True):
+@handle_nestable
+def gradient_descent_update(
+    ws: Union[ivy.Array, ivy.NativeArray],
+    dcdws: Union[ivy.Array, ivy.NativeArray],
+    lr: Union[int, List[int]],
```

Comment:
again, `float` not `int`, and are you sure about `List[int]`?

Reply:
No, my bad.